src/main/java/com/savitoh/casacodigoapi/model/Livro.java

```diff
@@ -0,0 +1,107 @@
+package com.savitoh.casacodigoapi.model;
+
+import javax.persistence.*;
+import javax.validation.constraints.*;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@Entity
+@Table
+@SequenceGenerator(name = "livro_seq", sequenceName = "livro_livro_id_seq", allocationSize = 1)
+public class Livro {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "livro_seq")
+    @Column(name = "livro_id")
+    private Integer livroId;
+
+    @Column(unique = true, length = 250)
+    @NotBlank
+    @Size(max = 250)
+    private String titulo;
+
+    @Column(length = 500)
+    @NotBlank
+    @Size(max = 500)
+    private String resumo;
+
+    @Lob
+    private String sumario;
+
+    @Column(precision = 6, scale = 2)
+    @NotNull
+    @Digits(integer=6, fraction=2)
+    @DecimalMin("20.00")
+    private BigDecimal preco;
+
+    @Column(name = "numero_paginas")
+    @NotNull
+    @Min(100)
+    private Integer numeroPaginas;
+
+    @Column(length = 17, unique = true)
+    @NotBlank
+    @Pattern(regexp = "^(?:ISBN(?:-1[03])?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})[- 0-9X]{13}$|97[89][0-9]{10}$|(?=(?:[0-9]+[- ]){4})[- 0-9]{17}$)(?:97[89][- ]?)?[0-9]{1,5}[- ]?[0-9]+[- ]?[0-9]+[- ]?[0-9X]$")
+    private String isbn;
+
+    @Column(name = "data_lancamento", columnDefinition = "TIMESTAMP")
+
+    @Future
+    @NotNull
+    private LocalDateTime dataLancamento;
+
+    @ManyToOne(optional = false, fetch = FetchType.LAZY)
+    @JoinColumn(name = "categoria_id")
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

O default para a geração da coluna da chave estrangeira já é o nome do atributo_atributoIdDaOutraEntidade. Neste caso já seria categoria_id. Se quiser manter o join column tudo bem, mas o efeito é o mesmo do default.

Reply:
Em relação ao optional, tudo bem. É mais uma garantia de borda, apesar que o NotNull já faria o papel.

*A complicação do nosso é código é proporcional a complicação da nossa feature. Quanto mais simples, melhor.*

Sobre o lazy... Essa é uma decisão que não é tão fácil. Você evitar um join pode ser bom, por outro lado se você for usar a categoria, você vai disparar 2 queries. Sugiro manter o comportamento default, que é deixar eager para relacionamentos **ToOne e aí ver o que vai rolar.